db.js

```diff
@@ -846,37 +876,50 @@ module.exports.CreateDB = function (parent, func) {
-        var dbname = (connectinArgs.database != null) ? connectinArgs.database : 'meshcentral';
-        delete connectinArgs.database;
-        obj.databaseType = 6;
-        const { Pool, Client } = require('pg');
-        connectinArgs.database = dbname;
+        let connectinArgs = parent.args.postgres;
+        connectinArgs.Database = (databaseName = (connectinArgs.database != null) ? connectinArgs.database : 'meshcentral');
+
+        let DatastoreTest;
+        obj.databaseType = DB_POSTGRESQL;
+        const { Client } = require('pg');
         Datastore = new Client(connectinArgs);
-        Datastore.connect();
-        sqlDbQuery('SELECT 1 FROM pg_database WHERE datname = $1', [dbname], function (dberr, dbdocs) { // check database exists first before creating
-            if (dberr == null) { // database exists now check tables exists
-                sqlDbQuery('SELECT doc FROM main WHERE id = $1', ['DatabaseIdentifier'], function (err, docs) {
-                    if (err == null) { setupFunctions(func); } else { postgreSqlCreateTables(func); } // If not present, create the tables and indexes
+        //Connect to and check pg db first to check if own db exists. Otherwise errors out on 'database does not exist'
+        connectinArgs.database = 'postgres';
+        DatastoreTest = new Client(connectinArgs);
+        DatastoreTest.connect();
+
+        DatastoreTest.query('SELECT 1 FROM pg_catalog.pg_database WHERE datname = $1', [databaseName], function (err, res) { // check database exists first before creating
+            if (res.rowCount != 0) { // database exists now check tables exists
+                Datastore.connect();
+                DatastoreTest.end();
+                Datastore.query('SELECT doc FROM main WHERE id = $1', ['DatabaseIdentifier'], function (err, res) {
+                    if (err == null) {
+                      (res.rowCount ==0) ? postgreSqlCreateTables(func) : setupFunctions(func)
+                    } else
+                    if (err.code == '42P01') { //42P01 = undefined table, https://www.postgresql.org/docs/current/errcodes-appendix.html
+                        postgreSqlCreateTables(func);
+                    } else {
+                        console.log('Postgresql database exists, other error: ', err.message); process.exit(0);
+                    };
                 });
             } else { // If not present, create the tables and indexes
-                const pgtools = require('pgtools');
-                pgtools.createdb(connectinArgs, dbname, function (err, res) {
+                //not needed, just use a create db statement: const pgtools = require('pgtools'); 
+                DatastoreTest.query('CREATE DATABASE '+ databaseName + ';', [], function (err, res) {
                     if (err == null) {
                         // Create the tables and indexes
+                        Datastore.connect();
+                        DatastoreTest.end();
```

Comment:
surely it would be best to end the test connection before opening a new one to the same database?
otherwise we would have 2 connections open a single point in time, even if its for a few miliseconds

Reply:
Ah yes, done.